lib/portlayer/network/context.go

```diff
@@ -940,7 +940,13 @@ func (c *Context) RemoveIDFromScopes(op trace.Operation, id string) ([]*Endpoint
 		s := ne.scope
 
 		// save the endpoint info
-		e := con.Endpoint(s).copy()
+		e := con.Endpoint(s)
+
+		if e == nil {
```

Comment:
panic'ing in this circumstance is the correct thing to do. It's indicative of a violation of our core assumptions.

There are two things that should occur given we're seeing cascade failures from this:
1. fix the root cause
2. fix the fact that the personality doesn't handle portlayer restarts

To address part of (2) we can/should just have the personality re-establish the event stream if it dies. That's partial.
The more complex part is re-validating cache state, etc which should occur (I hate multi-level caches without true invalidation/revalidation schemes)